firmware/common/ui_widget.cpp

```diff
@@ -1524,6 +1524,24 @@ void OptionsField::set_by_value(value_t v) {
     set_selected_index(0);
 }
 
+void OptionsField::set_by_nearest_value(value_t v) {
```

Comment:
Needed this so that you could pass step values between apps and have them snap to the correct step value in the UI.